File: 19-remove-nth-node-from-end-of-list/19-remove-nth-node-from-end-of-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        ListNode* temp=new ListNode();
        temp->next=head;
        ListNode* p1=temp;
        ListNode* p2=temp;
        while (n>0)
        {
            p1=p1->next;
            n--;
        }
        while (p1!=nullptr && p1->next!=nullptr)
        {
            p1=p1->next;
            p2=p2->next;
        }
        p2->next=p2->next->next;
        return temp->next;
    }
};

File: 19-remove-nth-node-from-end-of-list/19-remove-nth-node-from-end-of-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def removeNthFromEnd(self, head: ListNode, n: int) -> ListNode:
        p1=p2=head
        while n!=0 and p1.next:
            n-=1
            p1=p1.next
        if n!=0:
            return head.next
            
        while p1 and p1.next:
            p1=p1.next
            p2=p2.next
        if p2 and p2.next:
            p2.next=p2.next.next
        return head